domain/consensus/processes/syncmanager/antipast.go

```diff
@@ -110,60 +110,41 @@ func (sm *syncManager) missingBlockBodyHashes(highHash *externalapi.DomainHash)
 	}
 
 	lowHash := headerTipsPruningPoint
+	foundHeaderOnlyBlock := false
 	for selectedChildIterator.Next() {
 		selectedChild := selectedChildIterator.Get()
 		selectedChildStatus, err := sm.blockStatusStore.Get(sm.databaseContext, selectedChild)
 		if err != nil {
 			return nil, err
 		}
 
-		if selectedChildStatus != externalapi.StatusHeaderOnly {
-			lowHash = selectedChild
+		if selectedChildStatus == externalapi.StatusHeaderOnly {
+			foundHeaderOnlyBlock = true
 			break
 		}
+		lowHash = selectedChild
 	}
-
-	hashesBetween, err := sm.antiPastHashesBetween(lowHash, highHash)
-	if err != nil {
-		return nil, err
+	if !foundHeaderOnlyBlock {
```

Comment:
Can you try to revert it to use selectedChildIterator but use reachability's children instead?

Reply:
I tried to, but unfortunately, I still got wrong results.